controllers/provisioners/config.go

```diff
@@ -189,6 +190,12 @@ func (c *ProvisionerConfiguration) setRestrictedFields(unstructuredInstanceGroup
 			if err != nil {
 				errors.Wrap(err, "failed to set nested field")
 			}
+		} else {
+			// if no default value exist, make sure to ignore CR value for restricted fields
+			err := unstructured.SetNestedField(unstructuredInstanceGroup, nil, path...)
+			if err != nil {
+				errors.Wrap(err, "failed to set nested field to nil")
```

Comment:
@backjo I just noticed we are not returning any of these errors here.. do you remember if we did this intentionally? Thinking we should return these errors to caller WDYT

Reply:
:+1:  I don't think it was intentional